classes/class-wc-connect-compatibility-wc30.php

```diff
@@ -48,23 +51,32 @@ public function get_payment_method( WC_Order $order ) {
 		/**
 		 * Retrieve the corresponding Product for the given Order Item.
 		 *
-		 * @param WC_Order                                  $order
-		 * @param WC_Order_Item|WC_Order_Item_Product|array $item
+		 * @param WC_Order                                  $order WC Order.
+		 * @param WC_Order_Item|WC_Order_Item_Product|array $item  Order Item.
 		 *
-		 * @return WC_Product
+		 * @return WC_Product|bool
 		 */
 		public function get_item_product( WC_Order $order, $item ) {
 			if ( is_array( $item ) ) {
 				return wc_get_product( $item['product_id'] );
 			}
-			return $item->get_product();
+			if ( is_a( $item, 'WC_Order_Item_Product' ) ) {
```

Comment:
The phpdoc says this could be either `WC_Order_Item` or `WC_Order_Item_Product`. Should we check for `WC_Order_Item` as well?